base/show.jl

```diff
@@ -12,6 +12,15 @@ end
 
 show(io::IO, ::MIME"text/plain", r::AbstractRange) = show(io, r) # always use the compact form for printing ranges
 
+function show(io::IO, ::MIME"text/plain", r::UnitRange)
+    show(io, first(r))
+    print(io, ':')
+    show(io, last(r))
+    if !get(io, :compact, false) && isempty(r)
```

Comment:
I'd type assert `Bool` here just to be safe, because `typeof(io)` is often not known concretely at compilation time (`stdout` is a global variable that's not concretely typed). Might make some code less vulnerable to invalidation.

```suggestion
    if !(get(io, :compact, false)::Bool) && isempty(r)
```